src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2265 @@
+        """
+
+        return inputs.transpose(*self.dims)
+
+
+# TODO: add copied from after PatchTST master merge
+class PatchTSMixerBatchNorm(nn.Module):
+    """
+    Parameters:
+    Compute batch normalization
+        d_model (`int`): model dimension
+    """
+
+    def __init__(self, d_model):
+        super().__init__()
+        self.d_model = d_model
+        self.transpose = PatchTSMixerTranspose(1, 2)
+        self.batchnorm = nn.BatchNorm1d(self.d_model)
+
+    def forward(self, inputs: torch.Tensor):
+        """
+        Parameters:
+            inputs (`torch.Tensor` of shape `(batch_size, sequence_length, d_model)`):
+                input for Batch norm calculation
+        Returns:
+            `torch.Tensor`: tensor
+        """
+        output = self.transpose(inputs)  # output: (batch_size, d_model, sequence_length)
+        output = self.batchnorm(output)
+        output = self.transpose(output)  # output: (batch_size, sequence_length, d_model)
+        return output
+
+
+class PatchTSMixerNormLayer(nn.Module):
+    """Normalization block
+
+    Args:
+        config (`PatchTSMixerConfig`, *required*):
+            Configuration.
+    """
+
+    def __init__(self, config: PatchTSMixerConfig):
+        super().__init__()
+
+        self.config = config
+
+        if "batch" in config.norm_mlp.lower():
+            self.norm = PatchTSMixerBatchNorm(config.num_features)
+        else:
+            self.norm = nn.LayerNorm(config.num_features)
+
+    def forward(self, inputs: torch.Tensor):
+        """
+        Args:
+            inputs (`torch.Tensor` of shape `((batch_size, num_channels, num_patches, num_features))`):
+                Input to the normalization layer.
+        Returns:
+            `torch.Tensor` of shape `((batch_size, num_channels, num_patches, num_features))`
+        """
+        if "batch" in self.config.norm_mlp.lower():
+            # reshape the data
+            inputs_reshaped = torch.reshape(
+                inputs,
+                (
+                    inputs.shape[0] * inputs.shape[1],
+                    inputs.shape[2],
+                    inputs.shape[3],
+                ),
+            )  # inputs_reshaped: [batch_size*num_channels, num_patches, num_features]
+
+            inputs_reshaped = self.norm(
+                inputs_reshaped
+            )  # inputs_reshaped: [batch_size*num_channels, num_patches, num_features]
+            # put back data to the original shape
+
+            inputs = torch.reshape(
+                inputs_reshaped,
+                (
+                    inputs.shape[0],
+                    inputs.shape[1],
+                    inputs.shape[2],
+                    inputs.shape[3],
+                ),
+            )
+
+        else:
+            inputs = self.norm(inputs)
+
+        return inputs
+
+
+class PatchTSMixerMLP(nn.Module):
+    def __init__(self, in_features, out_features, expansion_factor, dropout, last_dropout=True):
```

Comment:
expansion factor and dropout are always taken from the config

Reply:
and last dropout does not seem to be passed to this module no?

Reply:
corrected